src/minify-plugin.ts

```diff
@@ -173,6 +191,20 @@ class ESBuildMinifyPlugin {
 			);
 		}));
 	}
+
+	private extractLegalComments(code: string, assetName: string): string[] | undefined {
+		// Search the first position of legal comments
+		const index = code.search(/^\/[/*]/m);
```

Comment:
Prefer against regex because it's slower than normal string searching. Possible to use `.lastIndexOf()`?

Reply:
Legal comments can be multiple and start with `/*` or `//` like this:

```js
//! Legal comment A
/*! Legal comment B */
```

So, I think the `/^\/[/*]/m` regex is necessary (slow though).

Reply:
Actually, I don't think that regex will cover it:

> A "legal comment" is considered to be any statement-level comment in JS or rule-level comment in CSS that contains `@license` or `@preserve` or that starts with `//!` or `/*!`. These comments are preserved in output files by default since that follows the intent of the original authors of the code.

Demo:
https://hyrious.me/esbuild-repl/?version=0.14.49&mode=transform&input=%2F%2F%21+inline%0A%2F%2F+inline+%40license+inline%0A%2F%2F+inline+%40preserve+inline%0A%2F*%21+block+*%2F%0A%2F*%21%0A%0Ablock%0A%0A*%2F%0A%0A%2F*%0Ablock%0A%40license%0A*%2F%0A%0A%2F*%0Ablock%0A%40preserve%0A*%2F%0A

Reply:
Ah, you're right. It seems complicated to make all patterns to search legal comments... 😓